### PR TITLE
Bump python versions up to 3.6 and 3.7

### PR DESCRIPTION
**Pull request by marcharper**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,9 @@
 
+dist: xenial
 language: python
 python:
-  - 3.5
```
Comment (drvinceknight) [inline .travis.yml]: Unless there's anything backwards incompatible with 3.5 that we'd be looking to include I'd prefer to keep it (and of course add 3.7).

(For example, on a purely selfish note, for my teaching I'd need to get anaconda reinstalled on all our lab machines if we dropped 3.5.)

File: setup.py [hunk 2]
```diff
@@ -23,15 +23,15 @@
     packages=["axelrod", "axelrod.strategies", "axelrod.data"],
     url="http://axelrod.readthedocs.org/",
     license="The MIT License (MIT)",
-    description="Reproduce the Axelrod iterated prisoners dilemma tournament",
+    description="Axelrod game theory library for the iterated prisoners dilemma",
```
Comment (drvinceknight) [inline setup.py]: :+1:

File: README.rst [hunk 3]
```diff
@@ -74,7 +74,8 @@ a peer reviewed paper introducing the library (22 authors).
 Installation
 ------------
 
-The library requires Python 3.5 or greater.
+The library requires Python 3.6 or greater. We typically support the two latest
```
Comment (drvinceknight) [inline README.rst]: I know we said that we'd support the two latest in the past but perhaps we keep previous versions until there's a particular feature we want to take advantage of? (Perhaps there already is and I'm forgetting.)

Comment (drvinceknight) [reply README.rst]: I've just seen https://github.com/Axelrod-Python/Axelrod/pull/1233#issuecomment-453702661 which seems like a good reason to me. Would it be ok to just bump here to 3.6 and not include the `We typically support the two latest...`?

File: .travis.yml (hunk 1 above)
Comment (drvinceknight) [reply .travis.yml]: Ignore this I've just seen https://github.com/Axelrod-Python/Axelrod/pull/1233#issuecomment-453702661

Comment (marcharper) [reply .travis.yml]: I'm fine with keeping 3.5 if you need it. We just have to use the comment form of type hint, which we've already done in some cases.

Comment (drvinceknight) [reply .travis.yml]: If it's easy enough it could be nicer to keep things a bit more backwards compatible but I'm happy with whatever you prefer.

**Comment by marcharper** [comment]
I'm trying 3.5 again. I had to bump up to xenial for Travis to get 3.7, we'll see if it gets 3.5 still.

**Comment by drvinceknight** [comment]
Looks like travis isn't immediately happy, if it's a choice between 3.5 and 3.7 I'd suggest moving forward with 3.7 :)

**Comment by marcharper** [comment]
Well I haven't figured out the necessary voodoo to make this work with both 3.5 and 3.7 at the same time, so I'll close for now until we decide to drop 3.5.

**Pull request closed without merging**